### PR TITLE
Memory Pair Game

### PR DESCRIPTION
**Pull request by ORybka**
### Memory Pair Game

[Demo](https://orybka.github.io/memory-pair-game/)
[Code base](https://github.com/ORybka/memory-pair-game)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/ORybka/memory-pair/main.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+function createCards() {
+  shuffledCards.forEach((el) => {
+    const card = document.createElement('div');
+    card.className = 'card-container';
+    card.dataset.cardValue = el;
+    card.innerHTML = `
+      <div class="flipper">
+        <div class="front"></div>
+        <div class="back">
+          <img src="assets/cards/${el}.jfif" alt="card" class="card-image"/>
```
Comment (babayK0) [inline submissions/ORybka/memory-pair/main.js:28]: it is better to store full `src` path in your array/object

File: submissions/ORybka/memory-pair/main.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+  const clickedCard = target.closest('.card-container');
+  movesNumber();
+
+  if (clickedCard && !clickedCard.classList.contains('flipped')) {
+    clickedCard.classList.add('flipped');
+    cardPairArray.push(clickedCard);
+  }
+
+  if (cardPairArray.length === 2) {
+    if (cardPairArray[0].dataset.cardValue === cardPairArray[1].dataset.cardValue) {
```
Comment (babayK0) [inline submissions/ORybka/memory-pair/main.js:47]: use destructuring please
`const [firstCard, secondCard] = cardPairArray;`

File: submissions/ORybka/memory-pair/main.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+function addAnimation() {
+  cardPairArray.forEach((card) => {
+    const image = card.querySelector('.card-image');
+    image.parentNode.removeChild(image);
+    card.querySelector('.back').classList.add('fire');
+  });
+}
+
+function movesNumber() {
```
Comment (babayK0) [inline submissions/ORybka/memory-pair/main.js:152]: function name must be verb

File: submissions/ORybka/memory-pair/main.js [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+  document.querySelector('.moves').innerHTML = `Number of moves is <strong>${checkedPairs}</strong>`;
+}
+
+function ifWon() {
```
Comment (babayK0) [inline submissions/ORybka/memory-pair/main.js:156]: verb

File: submissions/ORybka/memory-pair/main.js [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+      cardsContainer.style.opacity = '0';
+      document.querySelector('header').style.opacity = '0';
+      document.querySelector('footer').style.opacity = '0';
+      setTimeout(() => {
+        wonMessage.style.display = 'block';
+        wonMessage.style.opacity = '1';
+        document.querySelector('header').style.opacity = '1';
+        document.querySelector('footer').style.opacity = '1';
+        document.querySelector('.main-container').style.overflow = 'hidden';
+      }, DELAY / 2);
```
Comment (babayK0) [inline submissions/ORybka/memory-pair/main.js:169]: wow, why not create css classes and just add them here ?

File: submissions/ORybka/memory-pair/main.js [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+    });
+    cardPairArray = [];
+    wonPair++;
+  }, DELAY * 1.5);
+  ifWon();
+}
+
+function addCat() {
+  if (window.matchMedia('(min-width: 768px)').matches) {
+    if (previousCat === '') {
```
Comment (babayK0) [inline submissions/ORybka/memory-pair/main.js:86]: `  if (!previousCat) {`

File: submissions/ORybka/memory-pair/main.js [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+    cardPairArray.forEach((card) => {
+      card.classList.add('hidden');
+    });
+    cardPairArray = [];
+    wonPair++;
+  }, DELAY * 1.5);
+  ifWon();
+}
+
+function addCat() {
```
Comment (babayK0) [inline submissions/ORybka/memory-pair/main.js:84]: Don't Repeat Yourself. So, make this function DRYer, please.

**Review by babayK0 — CHANGES_REQUESTED** [review]
Good job and very nice that you try to use js animations !
But watch my comments below to improve your code =)

**Comment by ORybka** [comment]
@babayK0 thank you very much for your comments 😸 I hope, I rewrote everything in the right way :)

**Review by babayK0 — APPROVED** [review]
:+1:

**Pull request merged**